mne/time_frequency/tfr.py

```diff
@@ -227,6 +227,237 @@ def _cwt(X, Ws, mode="same", decim=1, use_fft=True):
+        The number of good tapers (low-bias) is chosen automatically based on
+        this to equal floor(time_bandwidth - 1).
+    use_fft : bool, Defaults to True
+        Use the FFT for convolutions or not.
+    decim : int | slice, Defaults to 1
+        To reduce memory usage, decimation factor after time-frequency
+        decomposition.
+        If `int`, returns tfr[..., ::decim].
+        If `slice` returns tfr[..., decim].
+        Note that decimation may create aliasing artifacts.
+        Defaults to 1.
+    output : str, Defaults to 'complex'
+        'complex': single trial complex.
+        'power': single trial power.
+        'phase': single trial phase.
+        'avg_power': average of single trial power.
+        'avg_phaselock' phase locking factor across trials.
+        'avg_power_phaselock': average of single trial power and phase-locking factor across trials. # noqa
+    n_jobs : int, Defaults to 1
+        The number of epochs to process at the same time. The parallization is
+        implemented across channels.
+    verbose : bool, str, int, or None, Defaults to 'INFO'
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    out : array
+        Time frequency transform of epoch_data. If output is in ['complex',
+        'phase', 'power'], then shape of out is (n_epochs, n_chans, n_freqs,
+        n_times), else it is (n_chans, n_freqs, n_times). If output is
+        'avg_power_phaselock', the real values code for 'avg_power' and the
+        imaginary values code for the 'avg_phaselock':
+        out = avg_power + i * avg_phaselock
+    """
+    # Check data
+    epoch_data = np.asarray(epoch_data)
+    if epoch_data.ndim != 3:
+        raise ValueError('epoch_data must be of shape '
+                         '(n_epochs, n_chans, n_times)')
+
+    # Check frequencies
+    if not isinstance(frequencies, (list, np.ndarray)):
+        raise ValueError('frequencies must be an array-like.')
+    frequencies = np.asarray(frequencies, dtype=float)
+    if frequencies.ndim != 1:
+        raise ValueError('frequencies must be of shape n_freqs.')
+
+    # Check sfreq
+    if not isinstance(sfreq, (float, int)):
+        raise ValueError('sfreq must be a float or an int.')
+    sfreq = float(sfreq)
+
+    # Check params
+    allowed_ouput = ('complex', 'power', 'phase',
+                     'avg_power_phaselock', 'avg_power', 'avg_phaselock')
+    if output not in allowed_ouput:
+        raise ValueError("Unknown output type.")
+
+    if method not in ('mtm', 'morlet'):
+        raise ValueError('method must be "morlet" or "mtm"')
+
+    # Default zero_mean = True if mtm else False
+    zero_mean = method == 'mtm' if zero_mean is None else zero_mean
+    if not isinstance(zero_mean, bool):
+        raise ValueError('')
+    frequencies = np.asarray(frequencies)
+
+    # XXX Check not sure whether it makes sense to compute single-trial phases
+    # with mtm.
```

Comment:
Yes, this is what I realize while implementing it. But I'm not sure whether we should just throw a warning or actually prevent users doing it.

It wasn't possible to compute single trial phase with mtm in the previous code, so I guess we can leave this for another PR